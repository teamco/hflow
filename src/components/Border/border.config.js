/**
 * @constant
 * @type {{'0': number, '1': number, '2': number, '3': number}}
 */
const offsetWidth = {
  0: 85,
  1: 77,
  2: 68,
  3: 60
};

/**
 * @export
 * @param [props]
 * @return {{dims: {left: {width: string}, bottom: {color: string, width: string, style: string}, right: {width: *}}, direction: string}}
 */
export const topProps = (props = {}) => {
  const {
    direction = 'top',
    offset = 1,
    width = 15,
    height = 10,
    color = 'rgb(241, 241, 241)',
    style = 'solid'
  } = props;

  return {
    direction,
    dims: {
      left: { width: `${width}vw` },
      right: { width: `${offsetWidth[offset]}vw` },
      bottom: { width: `${height}vw`, color, style }
    }
  };
};

/**
 * @export
 * @param [props]
 * @return {{dims: {left: {width: *}, bottom: {color: string, width: string, style: string}, right: {width: string}}, direction: string}}
 */
export const bottomProps = (props = {}) => {
  const {
    direction = 'bottom',
    offset = 1,
    width = 15,
    height = 10,
    color = 'rgb(241, 241, 241)',
    style = 'solid'
  } = props;

  return {
    direction,
    dims: {
      left: { width: `${offsetWidth[offset]}vw` },
      right: { width: `${width}vw` },
      top: { width: `${height}vw`, color, style }
    }
  };
};

/**
 * @export
 * @param [props]
 * @return {`polygon(0% 0%, 100% 0, 100% ${*}%, ${number}% 100%, 0 ${*}%)`}
 */
export const bottomPolygon = (props = {}) => {
  const {
    offset = 1,
    width = 15
  } = props;

  const oWidth = offsetWidth[offset];

  return `polygon(0% 0%, 100% 0, 100% ${oWidth}%, ${100 - width}% 100%, 0 ${oWidth}%)`;
};