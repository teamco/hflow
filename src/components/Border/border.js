import React from 'react';
import classnames from 'classnames';

import styles from './border.module.less';

const Border = props => {
  const { className, direction, dims } = props;

  const types = {
    top: 'Bottom',
    bottom: 'Top',
    left: 'Right',
    right: 'Left'
  };

  const vertical = direction === 'top' || direction === 'bottom';

  let style = vertical ? {
    borderLeftWidth: dims.left.width,
    borderRightWidth: dims.right.width
  } : {
    borderTopWidth: dims.top.width,
    borderBottomWidth: dims.bottom.width
  };

  const route = types[direction];
  const routeKey = dims[route.toLowerCase()];

  style = {
    ...style,
    ...{
      [`border${route}Width`]: routeKey.width,
      [`border${route}Color`]: routeKey.color,
      [`border${route}Style`]: routeKey.style
    }
  };

  return (
      <div className={classnames(styles.border, styles[direction], className)}
           style={style}/>
  );
};

export default Border;
