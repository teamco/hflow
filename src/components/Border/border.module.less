.border {
  width: 0;
  height: 0;
}

.top, .bottom {
  border-left-color: transparent;
  border-left-style: solid;
  border-right-color: transparent;
  border-right-style: solid;
}

.right, .left {
  border-top-color: transparent;
  border-top-style: solid;
  border-bottom-color: transparent;
  border-bottom-style: solid;
}

