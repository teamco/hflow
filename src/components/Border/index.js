import Border from './border';

export default Border;
