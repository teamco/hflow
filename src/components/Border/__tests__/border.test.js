import React from 'react';
import { cleanup } from '@testing-library/react';

import { expectations } from '__tests__/helper';
import Border from '../border';

describe('@/components/Border', () => {

  // Note: running cleanup afterEach is done automatically for you in @testing-library/react@9.0.0 or higher
  // unmount and cleanup DOM after the test is finished.
  afterEach(cleanup);

  it('Top', () => {
    const topProps = {
      direction: 'top',
      dims: {
        left: { width: '15vw' },
        right: { width: '45vw' },
        bottom: { width: '15vh', color: 'rgb(241, 241, 241)', style: 'solid' }
      }
    };

    const _borderDom = expectations(Border, 'border', topProps);

    expect(_borderDom).toHaveClass('border top');
    expect(_borderDom).toHaveStyle(
        'border-left-width: 15vw; border-right-width: 45vw; border-bottom-width: 15vh; border-bottom-color: rgb(241, 241, 241); border-bottom-style: solid;');
  });

  it('Bottom', () => {
    const bottomProps = {
      direction: 'bottom',
      dims: {
        left: { width: '45vw' },
        right: { width: '15vw' },
        top: { width: '15vh', color: 'rgb(241, 241, 241)', style: 'solid' }
      }
    };

    const _borderDom = expectations(Border, 'border', bottomProps);

    expect(_borderDom).toHaveClass('border bottom');
    expect(_borderDom).toHaveStyle(
        'border-left-width: 45vw; border-right-width: 15vw; border-top-width: 15vh; border-top-color: rgb(241, 241, 241); border-top-style: solid;');
  });
});
