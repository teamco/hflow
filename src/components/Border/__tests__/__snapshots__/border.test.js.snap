// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@/components/Border Bottom 1`] = `
<DocumentFragment>
  <div
    class="border bottom"
    data-testid="border"
    style="border-left-width: 45vw; border-right-width: 15vw; border-top-width: 15vh; border-top-color: rgb(241, 241, 241); border-top-style: solid;"
  />
</DocumentFragment>
`;

exports[`@/components/Border Top 1`] = `
<DocumentFragment>
  <div
    class="border top"
    data-testid="border"
    style="border-left-width: 15vw; border-right-width: 45vw; border-bottom-width: 15vh; border-bottom-color: rgb(241, 241, 241); border-bottom-style: solid;"
  />
</DocumentFragment>
`;

exports[`components/Border Bottom 1`] = `
<DocumentFragment>
  <div
    class="border bottom"
    data-testid="border"
    style="border-left-width: 45vw; border-right-width: 15vw; border-top-width: 15vh; border-top-color: rgb(241, 241, 241); border-top-style: solid;"
  />
</DocumentFragment>
`;

exports[`components/Border Top 1`] = `
<DocumentFragment>
  <div
    class="border top"
    data-testid="border"
    style="border-left-width: 15vw; border-right-width: 45vw; border-bottom-width: 15vh; border-bottom-color: rgb(241, 241, 241); border-bottom-style: solid;"
  />
</DocumentFragment>
`;
